src/bucket.rs

```diff
@@ -84,6 +86,24 @@ where
         self.save(key, &output)?;
         Ok(output)
     }
+
+    /// may_update is like update, but can handle missing values:
+    /// * If there is no data at this key, the input is None
+    /// * We don't save data if the action returns None
+    ///
+    /// This is the least stable of the APIs, and definitely needs some usage
+    pub fn may_update(
+        &mut self,
+        key: &[u8],
+        action: &dyn Fn(Option<T>) -> Result<Option<T>>,
+    ) -> Result<Option<T>> {
```

Comment:
I like the API. It might as well be the primary one, since it does not treat missing values like an error. Hard to tell which API is desired.

Reply:
I prefer the last option above.

Option T to Result T

Using the Result Option T return value is a bit annoying and I haven't thought of a use for it

Reply:
Which last option? The `update` is T to Restult T, not Option T to Result T.

Reply:
For Singleton, we can assume the object exists before, for Bucket not so... maybe we just update the action for Bucket.update to |Option<T>| -> Result<T> and always support non-existent accounts, and always do a save. I think this would be better... but would be a breaking 0.2.0 release